diff -urN readline-4.2.org/terminal.c readline-4.2/terminal.c
--- readline-4.2.org/terminal.c	Wed Apr 11 19:45:39 2001
+++ readline-4.2/terminal.c	Wed Apr 11 19:51:14 2001
@@ -297,7 +300,7 @@
   { "im", &_rl_term_im },
   { "kd", &_rl_term_kd },
   { "kh", &_rl_term_kh },	/* home */
-  { "kH", &_rl_term_kH },	/* end */
+  { "@7", &_rl_term_kH },	/* end */
   { "kl", &_rl_term_kl },
   { "kr", &_rl_term_kr },
   { "ku", &_rl_term_ku },
