diff -urN readline-4.2.org/terminal.c readline-4.2/terminal.c
--- readline-4.2.org/terminal.c	Wed Apr 11 19:45:39 2001
+++ readline-4.2/terminal.c	Wed Apr 11 19:51:14 2001
@@ -138,7 +138,10 @@
 
 /* The key sequences sent by the Home and End keys, if any. */
 static char *_rl_term_kh;
-static char *_rl_term_kH;
+static char *_rl_term_kend;
+
+/* The key sequence sent by the Delete key, if any. */
+static char *_rl_term_kD;
 
 /* Variables that hold the screen dimensions, used by the display code. */
 int _rl_screenwidth, _rl_screenheight, _rl_screenchars;
@@ -297,7 +300,8 @@
   { "im", &_rl_term_im },
   { "kd", &_rl_term_kd },
   { "kh", &_rl_term_kh },	/* home */
-  { "kH", &_rl_term_kH },	/* end */
+  { "@7", &_rl_term_kend },	/* end */
+  { "kD", &_rl_term_kD },	/* delete */
   { "kl", &_rl_term_kl },
   { "kr", &_rl_term_kr },
   { "ku", &_rl_term_ku },
@@ -453,7 +457,8 @@
   _rl_bind_if_unbound (_rl_term_kl, rl_backward);
 
   _rl_bind_if_unbound (_rl_term_kh, rl_beg_of_line);	/* Home */
-  _rl_bind_if_unbound (_rl_term_kH, rl_end_of_line);	/* End */
+  _rl_bind_if_unbound (_rl_term_kend, rl_end_of_line);	/* End */
+  _rl_bind_if_unbound (_rl_term_kD, rl_delete);			/* Delete */       
 
 #if defined (VI_MODE)
   _rl_keymap = vi_movement_keymap;
@@ -463,7 +468,8 @@
   _rl_bind_if_unbound (_rl_term_kl, rl_backward);
 
   _rl_bind_if_unbound (_rl_term_kh, rl_beg_of_line);	/* Home */
-  _rl_bind_if_unbound (_rl_term_kH, rl_end_of_line);	/* End */
+  _rl_bind_if_unbound (_rl_term_kend, rl_end_of_line);	/* End */
+  _rl_bind_if_unbound (_rl_term_kD, rl_delete);			/* Delete */       
 #endif /* VI_MODE */
 
   _rl_keymap = xkeymap;
