diff -ru readline-4.0.old/Makefile.in readline-4.0/Makefile.in
--- readline-4.0.old/Makefile.in	Thu Feb 18 18:24:11 1999
+++ readline-4.0/Makefile.in	Thu Jun 24 19:29:29 1999
@@ -167,34 +167,35 @@
 
 install: installdirs $(STATIC_LIBS)
 	for f in ${INSTALLED_HEADERS}; do \
-		$(INSTALL_DATA) $(srcdir)/$$f $(includedir)/readline ; \
+		$(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(includedir)/readline ; \
 	done
-	-$(MV) $(libdir)/libreadline.a $(libdir)/libreadline.old
-	$(INSTALL_DATA) libreadline.a $(libdir)/libreadline.a
-	-test -n "$(RANLIB)" && $(RANLIB) -t $(libdir)/libreadline.a
-	-$(MV) $(libdir)/libhistory.a $(libdir)/libhistory.old
-	$(INSTALL_DATA) libhistory.a $(libdir)/libhistory.a
-	-test -n "$(RANLIB)" && $(RANLIB) -t $(libdir)/libhistory.a
+	-$(MV) $(libdir)/libreadline.a $(DESTDIR)$(libdir)/libreadline.old
+	$(INSTALL_DATA) libreadline.a $(DESTDIR)$(libdir)/libreadline.a
+	-test -n "$(RANLIB)" && $(RANLIB) -t $(DESTDIR)$(libdir)/libreadline.a
+	-$(MV) $(libdir)/libhistory.a $(DESTDIR)$(libdir)/libhistory.old
+	$(INSTALL_DATA) libhistory.a $(DESTDIR)$(libdir)/libhistory.a
+	-test -n "$(RANLIB)" && $(RANLIB) -t $(DESTDIR)$(libdir)/libhistory.a
 	-( if test -d doc ; then \
 		cd doc && \
 		${MAKE} ${MFLAGS} infodir=$(infodir) $@; \
 	  fi )
 
 installdirs: $(srcdir)/support/mkdirs
-	-$(SHELL) $(srcdir)/support/mkdirs $(includedir) \
-		$(includedir)/readline $(libdir) $(infodir) $(man3dir)
+	-$(SHELL) $(srcdir)/support/mkdirs $(DESTDIR)$(includedir) \
+		$(DESTDIR)$(includedir)/readline $(DESTDIR)$(libdir) \
+		$(DESTDIR)$(infodir) $(DESTDIR)$(man3dir)
 
 uninstall:
-	-test -n "$(includedir)" && cd $(includedir)/readline && \
+	-test -n "$(DESTDIR)$(includedir)" && cd $(DESTDIR)$(includedir)/readline && \
 		${RM} ${INSTALLED_HEADERS}
-	-test -n "$(libdir)" && cd $(libdir) && \
+	-test -n "$(DESTDIR)$(libdir)" && cd $(DESTDIR)$(libdir) && \
 		${RM} libreadline.a libreadline.old libhistory.a libhistory.old $(SHARED_LIBS)
 
 install-shared: installdirs shared
-	-( cd shlib ; ${MAKE} ${MFLAGS} install )
+	-( cd shlib ; ${MAKE} ${MFLAGS} DESTDIR=$(DESTDIR) install )
 
 uninstall-shared:
-	-( cd shlib; ${MAKE} ${MFLAGS} uninstall )
+	-( cd shlib; ${MAKE} ${MFLAGS} DESTDIR=$(DESTDIR) uninstall )
 
 TAGS:	force
 	$(ETAGS) $(CSOURCES) $(HSOURCES)
diff -ru readline-4.0.old/doc/Makefile.in readline-4.0/doc/Makefile.in
--- readline-4.0.old/doc/Makefile.in	Thu Feb 18 18:24:36 1999
+++ readline-4.0/doc/Makefile.in	Thu Jun 24 19:33:10 1999
@@ -113,26 +113,26 @@
 	$(RM) Makefile
 
 installdirs:	$(top_srcdir)/support/mkdirs
-	-$(SHELL) $(top_srcdir)/support/mkdirs $(infodir) $(man3dir)
+	-$(SHELL) $(top_srcdir)/support/mkdirs $(DESTDIR)$(infodir) $(DESTDIR)$(man3dir)
 
 install:	installdirs
 	if test -f readline.info; then \
-		${INSTALL_DATA} readline.info $(infodir)/readline.info; \
+		${INSTALL_DATA} readline.info $(DESTDIR)$(infodir)/readline.info; \
 	else \
-		${INSTALL_DATA} $(srcdir)/readline.info $(infodir)/readline.info; \
+		${INSTALL_DATA} $(srcdir)/readline.info $(DESTDIR)$(infodir)/readline.info; \
 	fi
 	if test -f history.info; then \
-		${INSTALL_DATA} history.info $(infodir)/history.info; \
+		${INSTALL_DATA} history.info $(DESTDIR)$(infodir)/history.info; \
 	else \
-		${INSTALL_DATA} $(srcdir)/history.info $(infodir)/history.info; \
+		${INSTALL_DATA} $(srcdir)/history.info $(DESTDIR)$(infodir)/history.info; \
 	fi
 	if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
-		install-info --dir-file=$(infodir)/dir $(infodir)/readline.info ; \
-		install-info --dir-file=$(infodir)/dir $(infodir)/history.info ; \
+		install-info --dir-file=$(DESTDIR)$(infodir)/dir $(DESTDIR)$(infodir)/readline.info ; \
+		install-info --dir-file=$(DESTDIR)$(infodir)/dir $(DESTDIR)$(infodir)/history.info ; \
 	else true; fi
-	-${INSTALL_DATA} $(srcdir)/readline.3 $(man3dir)/readline.3
+	-${INSTALL_DATA} $(srcdir)/readline.3 $(DESTDIR)$(man3dir)/readline.3
 
 uninstall:
-	$(RM) $(infodir)/readline.info
-	$(RM) $(infodir)/history.info
-	$(RM) $(man3dir)/readline.3
+	$(RM) $(DESTDIR)$(infodir)/readline.info
+	$(RM) $(DESTDIR)$(infodir)/history.info
+	$(RM) $(DESTDIR)$(man3dir)/readline.3
diff -ru readline-4.0.old/shlib/Makefile.in readline-4.0/shlib/Makefile.in
--- readline-4.0.old/shlib/Makefile.in	Thu Jun 24 19:22:20 1999
+++ readline-4.0/shlib/Makefile.in	Thu Jun 24 19:31:05 1999
@@ -144,16 +144,16 @@
 	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -lc -o $@ $(SHARED_HISTOBJ) xmalloc.so $(SHLIB_LIBS)
 
 installdirs: $(topdir)/support/mkdirs
-	-$(SHELL) $(topdir)/support/mkdirs $(libdir)
+	-$(SHELL) $(topdir)/support/mkdirs $(DESTDIR)$(libdir)
 
 install: installdirs $(SHLIB_STATUS)
-	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(libdir) -i "$(INSTALL_DATA)" $(SHARED_HISTORY)
-	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(libdir) -i "$(INSTALL_DATA)" $(SHARED_READLINE)
+	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(DESTDIR)$(libdir) -i "$(INSTALL_DATA)" $(SHARED_HISTORY)
+	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(DESTDIR)$(libdir) -i "$(INSTALL_DATA)" $(SHARED_READLINE)
 	@echo install: you may need to run ldconfig
 
 uninstall:
-	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(libdir) -U $(SHARED_HISTORY)
-	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(libdir) -U $(SHARED_READLINE)
+	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(DESTDIR)$(libdir) -U $(SHARED_HISTORY)
+	$(SHELL) $(topdir)/support/shlib-install -O $(host_os) -d $(DESTDIR)$(libdir) -U $(SHARED_READLINE)
 	@echo uninstall: you may need to run ldconfig
 
 clean mostlyclean:	force
