diff -urNp -x '*.orig' readline-8.1.org/readline.h readline-8.1/readline.h
--- readline-8.1.org/readline.h	2020-05-20 16:42:27.000000000 +0200
+++ readline-8.1/readline.h	2021-03-21 12:54:46.986931009 +0100
@@ -32,6 +32,7 @@ extern "C" {
 #  include "keymaps.h"
 #  include "tilde.h"
 #else
+#  include <stdio.h>
 #  include <readline/rlstdc.h>
 #  include <readline/rltypedefs.h>
 #  include <readline/keymaps.h>
