diff -urNp -x '*.orig' readline-8.1.org/config.h.in readline-8.1/config.h.in
--- readline-8.1.org/config.h.in	2016-11-21 19:02:48.000000000 +0100
+++ readline-8.1/config.h.in	2021-03-21 12:54:58.441931009 +0100
@@ -32,6 +32,9 @@
 
 #undef volatile
 
+/* Define for large files, on AIX-style hosts. */
+#undef _LARGE_FILES
+
 #undef PROTOTYPES
 #undef __PROTOTYPES
 
diff -urNp -x '*.orig' readline-8.1.org/histfile.c readline-8.1/histfile.c
--- readline-8.1.org/histfile.c	2020-03-17 17:37:35.000000000 +0100
+++ readline-8.1/histfile.c	2021-03-21 12:54:58.441931009 +0100
@@ -684,7 +684,7 @@ history_do_write (const char *filename,
   int file, mode, rv, exists;
   struct stat finfo, nfinfo;
 #ifdef HISTORY_USE_MMAP
-  size_t cursize;
+  off_t cursize;
 
   history_lines_written_to_file = 0;
 
