diff -Nru readline-4.0/shlib/Makefile.in readline-4.0.new/shlib/Makefile.in
--- readline-4.0/shlib/Makefile.in	Thu Feb 18 18:25:29 1999
+++ readline-4.0.new/shlib/Makefile.in	Tue Feb 23 06:47:33 1999
@@ -137,11 +137,11 @@
 
 $(SHARED_READLINE):	$(SHARED_OBJ)
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ) $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} @TERMCAP_LIB@ -lc -o $@ $(SHARED_OBJ) $(SHLIB_LIBS)
 
 $(SHARED_HISTORY):	$(SHARED_HISTOBJ) xmalloc.so
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_HISTOBJ) xmalloc.so $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -lc -o $@ $(SHARED_HISTOBJ) xmalloc.so $(SHLIB_LIBS)
 
 installdirs: $(topdir)/support/mkdirs
 	-$(SHELL) $(topdir)/support/mkdirs $(libdir)
