This comes from the bash package:

--- readline-5.0/display.c.read-e-segfault	2005-01-18 11:46:06.563478820 +0000
+++ readline-5.0/display.c	2005-01-18 11:47:08.791587924 +0000
@@ -335,7 +335,8 @@
   FREE (local_prompt_prefix);
 
   local_prompt = local_prompt_prefix = (char *)0;
-  prompt_last_invisible = prompt_visible_length = 0;
+  prompt_last_invisible = prompt_invis_chars_first_line = 0;
+  prompt_visible_length = prompt_physical_chars = 0;
 
   if (prompt == 0 || *prompt == 0)
     return (0);
