--- readline-8.1/aclocal.m4.orig	2021-09-30 12:19:56.932608638 +0200
+++ readline-8.1/aclocal.m4	2021-09-30 12:20:55.982231142 +0200
@@ -933,9 +933,9 @@ fi
 AC_CACHE_VAL(bash_cv_termcap_lib,
 [AC_CHECK_FUNC(tgetent, bash_cv_termcap_lib=libc,
   [AC_CHECK_LIB(termcap, tgetent, bash_cv_termcap_lib=libtermcap,
+   [AC_CHECK_LIB(ncursesw, tgetent, bash_cv_termcap_lib=libncursesw,
     [AC_CHECK_LIB(tinfo, tgetent, bash_cv_termcap_lib=libtinfo,
         [AC_CHECK_LIB(curses, tgetent, bash_cv_termcap_lib=libcurses,
-	    [AC_CHECK_LIB(ncursesw, tgetent, bash_cv_termcap_lib=libncursesw,
                 [AC_CHECK_LIB(ncurses, tgetent, bash_cv_termcap_lib=libncurses,
 	            bash_cv_termcap_lib=gnutermcap)])])])])])])
 if test "X$_bash_needmsg" = "Xyes"; then
