--- readline-4.3/terminal.c.orig	Mon Mar  4 18:23:09 2002
+++ readline-4.3/terminal.c	Sun Jul 28 16:09:59 2002
@@ -143,8 +143,12 @@
 /* The key sequences sent by the Home and End keys, if any. */
 static char *_rl_term_kh;
 static char *_rl_term_kH;
+static char *_rl_term_kend;
 static char *_rl_term_at7;	/* @7 */
 
+/* The key sequence sent by the Delete key, if any. */
+static char *_rl_term_kD;
+
 /* Insert key */
 static char *_rl_term_kI;
 
@@ -313,7 +317,8 @@
   { "ei", &_rl_term_ei },
   { "ic", &_rl_term_ic },
   { "im", &_rl_term_im },
-  { "kH", &_rl_term_kH },	/* home down ?? */
+  { "@7", &_rl_term_kend },	/* end */
+  { "kD", &_rl_term_kD },	/* delete */
   { "kI", &_rl_term_kI },	/* insert */
   { "kd", &_rl_term_kd },
   { "ke", &_rl_term_ke },	/* end keypad mode */
@@ -495,7 +500,8 @@
   _rl_bind_if_unbound (_rl_term_kl, rl_backward);
 
   _rl_bind_if_unbound (_rl_term_kh, rl_beg_of_line);	/* Home */
-  _rl_bind_if_unbound (_rl_term_at7, rl_end_of_line);	/* End */
+  _rl_bind_if_unbound (_rl_term_kend, rl_end_of_line);	/* End */
+  _rl_bind_if_unbound (_rl_term_kD, rl_delete);		/* Delete */       
 
   _rl_keymap = xkeymap;
 }
