diff -urN readline-4.2.org/readline.h readline-4.2/readline.h
--- readline-4.2.org/readline.h	Sun Jun 17 20:14:22 2001
+++ readline-4.2/readline.h	Sun Jun 17 20:20:09 2001
@@ -32,6 +32,7 @@
 #  include "keymaps.h"
 #  include "tilde.h"
 #else
+#  include <stdio.h>
 #  include <readline/rlstdc.h>
 #  include <readline/keymaps.h>
 #  include <readline/tilde.h>
diff -urN readline-4.2.org/rlstdc.h readline-4.2/rlstdc.h
--- readline-4.2.org/rlstdc.h	Sun Jun 17 20:14:22 2001
+++ readline-4.2/rlstdc.h	Sun Jun 17 20:20:16 2001
@@ -22,6 +22,10 @@
 #if !defined (_RL_STDC_H_)
 #define _RL_STDC_H_
 
+#if defined (__cplusplus) && defined (__P) 
+#  undef __P 
+#endif 
+
 /* Adapted from BSD /usr/include/sys/cdefs.h. */
 
 /* A function can be defined using prototypes and compile on both ANSI C
