--- readline-8.1/aclocal.m4.orig	2021-09-30 12:19:56.932608638 +0200
+++ readline-8.1/aclocal.m4	2021-09-30 12:20:55.982231142 +0200
@@ -996,10 +996,10 @@
 AC_CACHE_VAL(bash_cv_termcap_lib,
 [AC_CHECK_FUNC(tgetent, bash_cv_termcap_lib=libc,
   [AC_CHECK_LIB(termcap, tgetent, bash_cv_termcap_lib=libtermcap,
+                [AC_CHECK_LIB(ncursesw, tgetent, bash_cv_termcap_lib=libncursesw,
     [AC_CHECK_LIB(tinfo, tgetent, bash_cv_termcap_lib=libtinfo,
         [AC_CHECK_LIB(curses, tgetent, bash_cv_termcap_lib=libcurses,
 	    [AC_CHECK_LIB(ncurses, tgetent, bash_cv_termcap_lib=libncurses,
-                [AC_CHECK_LIB(ncursesw, tgetent, bash_cv_termcap_lib=libncursesw,
 	            bash_cv_termcap_lib=gnutermcap)])])])])])])
 if test "X$_bash_needmsg" = "Xyes"; then
 AC_MSG_CHECKING(which library has the termcap functions)
@@ -1012,6 +1012,9 @@
 elif test $bash_cv_termcap_lib = libtermcap && test -z "$prefer_curses"; then
 TERMCAP_LIB=-ltermcap
 TERMCAP_DEP=
+elif test $bash_cv_termcap_lib = libncursesw; then
+TERMCAP_LIB=-lncursesw
+TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libtinfo; then
 TERMCAP_LIB=-ltinfo
 TERMCAP_DEP=
--- readline-8.1/configure.ac.orig	2021-09-30 12:14:21.587987979 +0200
+++ readline-8.1/configure.ac	2021-09-30 12:21:23.945384363 +0200
@@ -216,6 +216,7 @@
 -ltinfo)  TERMCAP_PKG_CONFIG_LIB=tinfo ;;
 -lcurses) TERMCAP_PKG_CONFIG_LIB=ncurses ;;
 -lncurses) TERMCAP_PKG_CONFIG_LIB=ncurses ;;
+-lncursesw) TERMCAP_PKG_CONFIG_LIB=ncursesw ;;
 -ltermcap) TERMCAP_PKG_CONFIG_LIB=termcap ;;
 *) TERMCAP_PKG_CONFIG_LIB=termcap ;;
 esac
