diff -Nur readline-4.2.orig/doc/hist.texinfo readline-4.2/doc/hist.texinfo
--- readline-4.2.orig/doc/hist.texinfo	Fri Feb  2 17:37:12 2001
+++ readline-4.2/doc/hist.texinfo	Sat Aug  4 13:11:15 2001
@@ -9,9 +9,9 @@
 @include manvers.texinfo
 
 @ifinfo
-@dircategory Libraries
+@dircategory Libraries:
 @direntry
-* History: (history).       The GNU history library API
+* History: (history).			The GNU history library API
 @end direntry
 
 This document describes the GNU History library, a programming tool that
diff -Nur readline-4.2.orig/doc/rlman.texinfo readline-4.2/doc/rlman.texinfo
--- readline-4.2.orig/doc/rlman.texinfo	Fri Feb  2 17:49:17 2001
+++ readline-4.2/doc/rlman.texinfo	Sat Aug  4 13:11:15 2001
@@ -9,9 +9,9 @@
 @include manvers.texinfo
 
 @ifinfo
-@dircategory Libraries
+@dircategory Libraries:
 @direntry
-* Readline: (readline).       The GNU readline library API
+* Readline: (readline).			The GNU readline library API
 @end direntry
 
 This document describes the GNU Readline Library, a utility which aids
diff -Nur readline-4.2.orig/doc/rluserman.texinfo readline-4.2/doc/rluserman.texinfo
--- readline-4.2.orig/doc/rluserman.texinfo	Fri Feb  2 17:59:01 2001
+++ readline-4.2/doc/rluserman.texinfo	Sat Aug  4 13:12:31 2001
@@ -8,9 +8,9 @@
 @include manvers.texinfo
 
 @ifinfo
-@dircategory Libraries
+@dircategory Libraries:
 @direntry
-* RLuserman: (rluserman).       The GNU readline library User's Manual.
+* RLuserman: (rluserman).		The GNU readline library User's Manual.
 @end direntry
 
 This document describes the end user interface of the GNU Readline Library,
